c/new-io.c

```diff
@@ -268,38 +259,28 @@ ptr S_compress_input_fd(INT fd, I64 pos) {
   return MAKE_FD(fd);
 }
 
-ptr S_compress_output_fd(INT fd, IBOOL as_gz) {
+ptr S_compress_output_fd(INT fd) {
   glzFile file;
-  int as_append;
-
-#ifdef WIN32
-  as_append = 0;
-#else
-  as_append = fcntl(fd, F_GETFL) & O_APPEND;
-#endif
+  ptr tc = get_thread_context();
 
-  if (as_gz)
-    file = glzdopen_gz(fd, as_append ? "ab" : "wb");
-  else
-    file = glzdopen_lz4(fd, as_append ? "ab" : "wb");
+  file = glzdopen_output(fd, (int)UNFIX(COMPRESSFORMAT(tc)), (int)UNFIX(COMPRESSLEVEL(tc)));
```

Comment:
The glzdopen_output function prototype uses INT instead of int, so wouldn't you want the casts to use INT instead of int also?